### PR TITLE
[MO] - [clients] -> removing future + solving issue with vertx

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR adding ability that clients only returning int type, which is sent or receive messages and encapsulate all the async calls inside the methods. 

Additionally, I have done some refactoring which was needed for instance:
 -  use instance attribute of `Vert.x` from the class AbstractVerticle
 -  auto-closeable interfaces, which ensures that the client will always close the `Vert.x` instance

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by see-quick** [comment]
@strimzi-ci run tests tests_only

**Comment by see-quick** [comment]
@strimzi-ci run tests profile=externalclients tests_only

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/Consumer.java [hunk 1]
```diff
@@ -13,8 +14,8 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.IntPredicate;
 
-
-public class Consumer extends ClientHandlerBase<Integer> {
+// TODO: autocloseable to close vertx always...
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/Consumer.java:17]: Do we still need the TODO?

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 2]
```diff
@@ -1290,11 +1286,11 @@ void testNodePortTls() throws Exception {
...
-        Future<Integer> producer = basicExternalKafkaClient.sendMessagesTls();
-        Future<Integer> consumer = basicExternalKafkaClient.receiveMessagesTls();
-
-        assertThat(producer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
-        assertThat(consumer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
+//        Future<Integer> producer = basicExternalKafkaClient.sendMessagesTls();
+//        Future<Integer> consumer = basicExternalKafkaClient.receiveMessagesTls();
+//
+//        assertThat(producer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
+//        assertThat(consumer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java:1293]: can we remove these lines?

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 3]
```diff
@@ -1324,11 +1320,11 @@ void testLoadBalancer() throws Exception {
...
-        Future<Integer> producer = basicExternalKafkaClient.sendMessagesPlain();
-        Future<Integer> consumer = basicExternalKafkaClient.receiveMessagesPlain();
-
-        assertThat(producer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
-        assertThat(consumer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
+//        Future<Integer> producer = basicExternalKafkaClient.sendMessagesPlain();
+//        Future<Integer> consumer = basicExternalKafkaClient.receiveMessagesPlain();
+//
+//        assertThat(producer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
+//        assertThat(consumer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java:1327]: ditto as above?

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/Consumer.java (hunk 1 above)
Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/Consumer.java:17]: No, we do not. I will remove it ;)

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java (hunk 2 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java:1293]: yeah, good point!!

**Comment by see-quick** [comment]
@strimzi-ci run tests tests_only

**Comment by see-quick** [comment]
@mstruk WDYT about this change. I guess I have done, what you have requested but I am free to improve it if you have any special commends.

**Pull request merged**